Tests/PackageGraphTests/DependencyResolverTests.swift

```diff
@@ -376,6 +382,38 @@ class DependencyResolverTests: XCTestCase {
         }
     }
 
+    func testPrefetchContainers() throws {
+        // Test that resolver requests for needed containers.
+        let provider = MockPackagesProvider(containers: [
+            MockPackageContainer(name: "A", dependenciesByVersion: [v1: [
+                (container: "AA", versionRequirement: v1Range),
+                (container: "AB", versionRequirement: v1Range),
+            ]]),
+            MockPackageContainer(name: "AA", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "AB", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "B", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "C", dependenciesByVersion: [v1: [
+                (container: "CA", versionRequirement: v1Range),
+            ]]),
+            MockPackageContainer(name: "CA", dependenciesByVersion: [v1: [
+                (container: "CAA", versionRequirement: v1Range),
+                (container: "CAB", versionRequirement: v1Range),
+            ]]),
+            MockPackageContainer(name: "CAA", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "CAB", dependenciesByVersion: [v1: []]),
+        ])
+        let resolver = MockDependencyResolver(provider, MockResolverDelegate(), enablePrefetching: true)
+
+        let result = try resolver.resolve(constraints: [
+            MockPackageConstraint(container: "A", versionRequirement: v1Range),
+            MockPackageConstraint(container: "B", versionRequirement: v1Range),
+            MockPackageConstraint(container: "C", versionRequirement: v1Range),
+        ])
+
+        XCTAssertEqual(result[0].version, v1)
+        XCTAssertEqual(provider.requestedContainers, ["A", "B", "C", "A", "B", "C", "AA", "AB", "AA", "AB", "CA", "CA", "CAA", "CAB", "CAA", "CAB"])
```

Comment:
hmm, I am not sure how to test the prefetching since we use the same method for prefetch + actual fetch